app/frontend/src/utils/transformUtils.js

```diff
@@ -64,3 +65,203 @@ export function attachAttributesToLinks(formSchemaComponents) {
     }
   });
 }
+
+
+/**
+ * @function isFormExpired
+ * Returns true for a form's schedule object if this form schedule available for that period
+ * @param {Object} form Schedule data
+ * @returns {boolean} TRUE if form not expired and Available to submission
+ * 
+ */
+export function isFormExpired(formSchedule = {}) {
+  var result = {
+    allowLateSubmissions:false,
+    expire:false,
+    message:''
+  };
+
+  if(formSchedule && formSchedule.enabled)
+  {
+    //Check if Form open date is in past or Is form already started for submission
+    if(formSchedule.openSubmissionDateTime){ 
+      var startDate = moment(formSchedule.openSubmissionDateTime).format('YYYY-MM-DD HH:MM:SS');
+      var isFormStartedAlready = moment().diff(startDate, 'seconds'); //If a positive number it means form get started
+      if(isFormStartedAlready >= 0){
+        //Form have valid past open date for scheduling so lets check for the next conditions
+        if(isFormStartedAlready && formSchedule.enabled){
+          if(formSchedule.closingMessage){
+            result = {...result,message:formSchedule.closingMessage};
+          }
+          var closeDate =  getCalculatedCloseSubmissionDate(
+            startDate,
+            formSchedule.keepOpenForTerm,
+            formSchedule.keepOpenForInterval,
+            formSchedule.allowLateSubmissions.enabled ? formSchedule.allowLateSubmissions.forNext.term : 0,
+            formSchedule.allowLateSubmissions.forNext.intervalType,
+            formSchedule.repeatSubmission.everyTerm,
+            formSchedule.repeatSubmission.everyIntervalType,
+            formSchedule.repeatSubmission.repeatUntil
+          ); //moment(formSchedule.closeSubmissionDateTime).format('YYYY-MM-DD HH:MM:SS');
+          var isBetweenStartAndCloseDate = moment().isBetween(startDate, closeDate);
+          
+          if(isBetweenStartAndCloseDate){
+            /** Check if form is Repeat enabled - start */
+            /** Check if form is Repeat enabled and alow late submition - start */
+            if(formSchedule.repeatSubmission.enabled){
+              var availableDates = getAvailableDates(
+                formSchedule.keepOpenForTerm,
+                formSchedule.keepOpenForInterval,
+                startDate,
+                formSchedule.repeatSubmission.everyTerm,
+                formSchedule.repeatSubmission.everyIntervalType,
+                formSchedule.allowLateSubmissions.enabled ? formSchedule.allowLateSubmissions.forNext.term : 0,
+                formSchedule.allowLateSubmissions.forNext.intervalType,
+                formSchedule.repeatSubmission.repeatUntil
+              );
+              for (let i = 0; i < availableDates.length; i++) {
```

Comment:
not really an issue since it passed the lint, but the spacing is not consistent before your open/close brackets

Reply:
Not sure Why is that showing there but when I did commit it was look good. More over on my local as well did not see any Lint Issues